File: Python/Labs/Lab_8/sort.py
def bubbleSort(x):
    if len(x) == 0:
        return [True, x]
    
    for i in x:
        if not type(x[0]) == type(i):
            return [False, x]
        
    swap = True
    while swap:
        swap = False
        for i in range (1, len(x), 1):
            if x[i - 1] > x[i]:
                temp = x[i]
                x[i] = x[i + 1]
                x[i + 1] = temp
                swap = True
                break
    return [True, x]

File: C/Labs/Lab_8/sort.c
#include<stdio.h>
#include<stdlib.h>

int bubbleSort(int *x, int size);

int bubbleSort(int *x, int size) {
    if (x == NULL || size == 0) {
        return -1;

    } else {

        int swap = 1, y = 0;
        while (swap) {
            swap = 0;
            int i = 0;
            for(i = 1; i < size; i++) {
                if (x[i - 1] > x[i]) {
                    y = x[i];
                    x[i] = x[i - 1];
                    x[i - 1] = y;
                    swap = 1;
                    break;
                }
            }
        }
        return 0;
    }
}